### ISSUE TITLE
Project doesn't properly load, files in project aren't recognized, and autocomplete doesn't work at all

### ISSUE THREAD

**Issue by sandoichi** [fsharp/emacs-fsharp-mode#166]
### Description

I'm using the F# layer with spacemacs.

When opening any F# file, I can see a message stating that the fsproj is loaded, but then I immediately get a message saying that my current file it not part of the loaded projects.

Error checking works on built-in .net libraries but nothing else. Modules from other files in project are not recognized and I also don't get any auto completion (not even on .net built ins).

I have msbuild 15 on my path. I'm running spacemacs (emacs 25.3.1) on windows (Tried on both windows7 and windows10 and both have the same issue).

### Repro steps

Please provide the steps required to reproduce the problem

1. Open up a .fs file

2. Get message that .fsproject is loaded

3. Get another messaging saying the current .fs file is not a part of the loaded projects.

### Expected behavior

I expect the project to be properly loaded, modules in the project to be recognized, and autocomplete to work.

### Actual behavior

Auto complete doesn't work.

Modules defined in the project scope are not recognized when imported.

```
Loaded F# project ’ConsoleApplication1.fsproj’
Error: c:/Users/sandoichi/dev/fsharp/ConsoleApplication1/ConsoleApplication1/Program.fs is not part of the loaded projects.

```

### Related information 

* Windows 10 and 7
* Master branch
* Emacs 25.3.1 (with spacemacs)
* .NET 4.5, 4.5.1, 4.6.1  / MSBuild 2017/2015/2013

**Comment by Gastove** [comment]
I'm seeing two different versions of this myself. For projects created with the never newest forge, I simply get, `The project was not restored.` -- certainly not the most... illuminating error. For older projects, I'm seeing this stack trace in the `*fsharp-debug*` buffer:

```
21:03:03.054357: {"Kind":"project","Data":{"Project":"/home/rdonaldson/Code/gearheart/src/Gift/Gift.fsproj","Files":[],"Output":"null","References":[],"Logs":{"/home/rdonaldson/Code/gearheart/src/Gift/Gift.fsproj":"Microsoft.Build.Exceptions.InvalidProjectFileException: /home/rdonaldson/Code/gearheart/src/Gift/Gift.fsproj: Object reference not set to an instance of an object\n  at Microsoft.FSharp.Compiler.SourceCodeServices.ProjectCrackerTool.ProjectCrackerTool+FSharpProjectFileInfo..ctor (System.String fsprojFileName, Microsoft.FSharp.Core.FSharpOption`1[T] properties, Microsoft.FSharp.Core.FSharpOption`1[T] enableLogging) [0x00129] in <599f46db06d737baa7450383db469f59>:0 \n  at Microsoft.FSharp.Compiler.SourceCodeServices.ProjectCrackerTool.ProjectCrackerTool.getOptions@387 (System.Boolean enableLogging, Microsoft.FSharp.Collections.FSharpList`1[T] properties, System.String file) [0x0000f] in <599f46db06d737baa7450383db469f59>:0 \n  at Microsoft.FSharp.Compiler.SourceCodeServices.ProjectCrackerTool.ProjectCrackerTool.getOptions (System.String file, System.Boolean enableLogging, Microsoft.FSharp.Collections.FSharpList`1[T] properties) [0x00005] in <599f46db06d737baa7450383db469f59>:0 \n  at Microsoft.FSharp.Compiler.SourceCodeServices.ProjectCrackerTool.ProjectCrackerTool.crackOpen (System.String[] argv) [0x000cc] in <599f46db06d737baa7450383db469f59>:0 "},"OutputType":"exe","Info":{"SdkType":"verbose","Data":{}},"AdditionalInfo":{}}}
```

Not, uh, a lot more illuminating, but at least.... it's more.

**Comment by Gastove** [comment]
Huh. Seems like there's every possibility that both old and new projects defined with Forge are just incompatible with current xbuild. That's cool. At least in my case, might not be an emacs-fsharp problem at all.

**Comment by sandoichi** [comment]
I should note that in my case, I am using a project created by Visual Studio on Windows. I've tested with various projects defined across several versions of visual studio (2013/2015/2017)

**Comment by juergenhoetzel** [comment]
> I'm seeing two different versions of this myself. For projects created with the never newest forge, I simply get, The project was not restored. -- certainly not the most... illuminating error.

The new [flycheck-verify](https://github.com/fsharp/emacs-fsharp-mode/pull/170) functionality should improve the user experience. 

@sandoichi  Could you eval in an elisp buffer:
```elisp
(pp fsharp-ac--project-files)
```

**Comment by sandoichi** [comment]
@juergenhoetzel 

Is this what you are looking for? (sorry, I'm an emacs noob)

```

#s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8125 data
())
"#s(hash-table size 65 test equal rehash-size 1.5 rehash-threshold 0.8125 data
())
"
evil-line-move: End of buffer
```

**Comment by juergenhoetzel** [comment]
Fixed in 1.9.3
